shanoir-ng-datasets/src/main/java/org/shanoir/ng/datasetacquisition/model/mr/MrSequenceVariant.java

```diff
@@ -95,7 +95,11 @@ public static MrSequenceVariant getIdByType(final String type) {
 		if (TOF.equals(type)) { // see GitHub issue #561
 			return MrSequenceVariant.SS;
 		}
-		return MrSequenceVariant.valueOf(type);
+		try {
+			return MrSequenceVariant.valueOf(type);
+		} catch (IllegalArgumentException exception) {
+			return UNKNOWN;
```

Comment:
You could use EnumUtils from apache-commons :
``` java
return EnumUtils.isValidEnum(MrSequenceVariant.class, type) : MrSequenceVariant.valueOf(type) ? UNKNOWN
```
The `isValidEnum` method use the same try-catch method as your solution, though.